packages/utils/src/managers/MaxZIndexManager.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { FC, useMemo } from 'react';
+
+
+export interface IMaxZIndexManager {
+  MaxZIndexContext: React.Context<number>;
+  Provider: FC;
+}
+
+export const getMaxZIndex = (): number | undefined => Array.from(document.querySelectorAll('body *'))
+  .map((node: Element) => parseFloat(window.getComputedStyle(node).zIndex))
+  .filter((zIndex: number) => !Number.isNaN(zIndex))
+  .sort((a, b) => a - b)
+  .pop();
+
+export const maxZIndexManager = (): IMaxZIndexManager => {
+  const MaxZIndexContext = React.createContext(1);
+
+  const Provider: FC = ({
```

Comment:
Provider -> MaxZIndexProvider

Reply:
It doesn't important, because we get Provider in other components with renaming, for example, in Modal component:

const { Provider: ModalProvider, MaxZIndexContext } = maxZIndexManager();